File: tasks/quadraticEquation.java
ArrayList<Double> quadraticEquation(int a, int b, int c) {

  int discriminant =  b*b - 4 * a * c;
  if (discriminant < 0) {
    ArrayList<Double> empty = new ArrayList<>();
    return empty;
  }
  if (discriminant == 0) {
    ArrayList<Double> answer = new ArrayList<>();
    answer.add(-b / (2.0 * a));
    return answer;
  }
  ArrayList<Double> roots = new ArrayList<>();
  roots.add((-b - Math.sqrt(discriminant) ) / (2.0 * a));
  roots.add((-b + Math.sqrt(discriminant) ) / (2.0 * a));
  if (roots.get(0) > roots.get(1)) {
    double tmp = roots.get(1);
    roots.set(1, roots.get(0));
    roots.set(0, tmp);
  }
  return roots;
}

File: tasks/quadraticEquation.js
function quadraticEquation(a, b, c) {

  var discriminant = b * b - 4 * a * c;
  if (discriminant < 0) {
    return [];
  }
  if (discriminant === 0) {
    return [-b / (2 * a)];
  }
  var roots = [];
  roots.push((-b - Math.sqrt(discriminant)) / (2 * a));
  roots.push((-b + Math.sqrt(discriminant)) / (2 * a));
  if (roots[0] > roots[1]) {
    var tmp = roots[1];
    roots[1] = roots[0];
    roots[0] = tmp;
  }
  return roots;
}
